packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -243,22 +243,28 @@ export const usePickerValue = <
     onSelectedSectionsChange,
   } = props;
 
-  const [value, setValue] = useControlled({
+  const utils = useUtils<TDate>();
+  const adapter = useLocalizationContext<TDate>();
+
+  const [rawValue, setValue] = useControlled({
     controlled: inValue,
     default: defaultValue ?? valueManager.emptyValue,
     name: 'usePickerState2',
     state: 'value',
   });
 
+  const value = React.useMemo(
```

Comment:
I forgot to apply this change on the new pickers when doing #6170 and it causes a bug on the range picker when the date is not valid